Shopify/cli
# Don't allow the log file to grow too much

### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify-cli-planning/issues/344

### WHAT is this pull request doing?
- Switching from loading in memory the whole logfile content to truncate the first lines to using streams so only little chunks are handled at a time
- This way each time a command is run the log is reduced to a max size, even if the current log file size is huge
- Not added the truncating process each time a message is written to the log to avoid overload.

### How to test your changes?
- Overwrite the existing log in ` ~/Library/Caches/shopify-cli-nodejs/shopify.cli.log` with a downloaded [sample huge csv file](https://eforexcel.com/wp/wp-content/uploads/2017/08/2m-Sales-Records.7z) and run any command. The log file should be truncated to a maximum (fixed) size similar to 5MB 

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
